File: ITapply/ITapply.Models/Requests/JobPostingUpdateRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static ITapply.Models.Responses.EnumResponse;

namespace ITapply.Models.Requests
{
    public class JobPostingUpdateRequest
    {
        [Required(ErrorMessage = "Job title is required.")]
        [StringLength(200, ErrorMessage = "Job title cannot exceed 200 characters.")]
        public string Title { get; set; }

        [Required(ErrorMessage = "Job description is required.")]
        [StringLength(10000, ErrorMessage = "Job description cannot exceed 10000 characters.")]
        public string Description { get; set; }

        [StringLength(5000, ErrorMessage = "Requirements cannot exceed 5000 characters.")]
        public string Requirements { get; set; }

        [StringLength(3000, ErrorMessage = "Benefits cannot exceed 3000 characters.")]
        public string Benefits { get; set; }

        [Required(ErrorMessage = "Employment type is required.")]
        public EmploymentType EmploymentType { get; set; }

        [Required(ErrorMessage = "Experience level is required.")]
        public ExperienceLevel ExperienceLevel { get; set; }

        [Range(1, int.MaxValue, ErrorMessage = "Location ID must be a positive number.")]
        public int? LocationId { get; set; }

        [Required(ErrorMessage = "Remote work option is required.")]
        public Remote Remote { get; set; }

        [Range(0, int.MaxValue, ErrorMessage = "Minimum salary must be a positive number or zero.")]
        public int MinSalary { get; set; }

        [Range(0, int.MaxValue, ErrorMessage = "Maximum salary must be a positive number or zero.")]
        public int MaxSalary { get; set; }

        [Required(ErrorMessage = "Application deadline is required.")]
        public DateTime ApplicationDeadline { get; set; }

        [Required(ErrorMessage = "Job posting status is required.")]
        public JobPostingStatus Status { get; set; }
        
        // List of skill IDs required for this job
        public List<int> SkillIds { get; set; } = new List<int>();
    }
} 

File: ITapply/ITapply.UI/itapply_mobile/lib/models/requests/job_posting_update_request.dart
import 'package:itapply_mobile/models/enums.dart';
import 'package:json_annotation/json_annotation.dart';

part 'job_posting_update_request.g.dart';

@JsonSerializable()
class JobPostingUpdateRequest {
  final String title;
  final String description;
  final String? requirements;
  final String? benefits;
  @JsonKey(toJson: _employmentTypeToJson)
  final EmploymentType employmentType;
  @JsonKey(toJson: _experienceLevelToJson)
  final ExperienceLevel experienceLevel;
  final int? locationId;
  @JsonKey(toJson: _remoteToJson)
  final Remote remote;
  final int? minSalary;
  final int? maxSalary;
  final DateTime applicationDeadline;
  @JsonKey(toJson: _jobPostingStatusToJson)
  final JobPostingStatus status;
  final List<int>? skillIds;

  JobPostingUpdateRequest({
    required this.title,
    required this.description,
    this.requirements,
    this.benefits,
    required this.employmentType,
    required this.experienceLevel,
    this.locationId,
    required this.remote,
    this.minSalary,
    this.maxSalary,
    required this.applicationDeadline,
    required this.status,
    this.skillIds,
  });
  
  factory JobPostingUpdateRequest.fromJson(Map<String, dynamic> json) => _$JobPostingUpdateRequestFromJson(json);
  Map<String, dynamic> toJson() => _$JobPostingUpdateRequestToJson(this);
}

int _employmentTypeToJson(EmploymentType type) => type.index;
int _experienceLevelToJson(ExperienceLevel level) => level.index;
int _remoteToJson(Remote remote) => remote.index;
int _jobPostingStatusToJson(JobPostingStatus status) => status.index;
